### PR TITLE
Test framework

### PR DESCRIPTION
**Pull request by Helco**
- [x] Testrunner on Windows, Linux, (MacOSX untested)
- [x] Tests for the test framework and types
- [x] Fixed `n_grect_standardize` and `n_grect_center_point`
- [x] Implement ressource loading/mapping
- [x] Implement (sub)screen comparing
- [x] Doxygen comments
- [x] Feedback to EMM (Evil Macro Magic)?
- [x] Change pointer style

### CODE REVIEW

File: contrib/testrunner_pc/resources.c [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+    return size_read;
+}
+
+ResImage* loadImageById(uint32_t resource_id) {
+    const char* name = getResourceNameById(resource_id);
+    return name == NULL ? NULL : loadImageByName(name);
+}
+
+ResImage* loadImageByName(const char* name) {
```
Comment (jwise) [inline contrib/testrunner_pc/resources.c]: `ResImage *loadImageByName`, `const char *name`

**Comment by jwise** [comment]
+1, looks good to me

File: contrib/testrunner_pc/resources.c (hunk 1 above)
Comment (Helco) [reply contrib/testrunner_pc/resources.c]: My colleagues don't call me "It's only a regex, right?" for nothing...

**Comment by jwise** [comment]
🎉

**Pull request merged**